lms/djangoapps/course_home_api/dates/v1/views.py

```diff
@@ -58,13 +60,16 @@ class DatesTabView(RetrieveAPIView):
 
     def get(self, request, *args, **kwargs):
         course_key_string = kwargs.get('course_key_string')
+        course_key = CourseKey.from_string(course_key_string)
+
+        if not course_home_mfe_dates_tab_is_active(course_key):
+            return Response(status=status.HTTP_404_NOT_FOUND)
```

Comment:
I think a slightly more "django-y" way to do this would be:
```
from django.http.response import Http404 
raise Http404
```
But that's a tiny nit. Your way is great too.